samplesheets/tests/test_views_taskflow.py

```diff
@@ -1829,6 +1837,385 @@ def test_accept_lock_failure(self):
         self._assert_alert_count(CREATE_ALERT, self.user_delegate, 1)
         self.assert_irods_obj(self.path, True)
 
+    def test_accept_collection(self):
+        """Test accepting a collection request with multiple objects inside"""
+        coll_path = os.path.join(self.assay_path, 'request_coll')
+        obj_path = os.path.join(coll_path, TEST_FILE_NAME)
+        self.irods.collections.create(coll_path)
+        self.irods.data_objects.create(obj_path)
+        self.assertEqual(self.irods.collections.exists(coll_path), True)
+        self.post_data['path'] = coll_path
+
+        with self.login(self.user_contrib):
+            self.client.post(
+                reverse(
+                    'samplesheets:irods_request_create',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+                self.post_data,
+            )
+
+        self.assertEqual(IrodsDataRequest.objects.count(), 1)
+        obj = IrodsDataRequest.objects.first()
+        self._assert_alert_count(CREATE_ALERT, self.user, 1)
+        self._assert_alert_count(CREATE_ALERT, self.user_delegate, 1)
+        self._assert_alert_count(ACCEPT_ALERT, self.user, 0)
+        self._assert_alert_count(ACCEPT_ALERT, self.user_delegate, 0)
+
+        with self.login(self.user):
+            response = self.client.post(
+                reverse(
+                    'samplesheets:irods_request_accept',
+                    kwargs={'irodsdatarequest': obj.sodar_uuid},
+                ),
+                {'confirm': True},
+            )
+            self.assertRedirects(
+                response,
+                reverse(
+                    'samplesheets:irods_requests',
+                    kwargs={'project': self.project.sodar_uuid},
+                ),
+            )
+            self.assertEqual(
+                list(get_messages(response.wsgi_request))[-1].message,
+                'iRODS data request "{}" accepted.'.format(
+                    obj.get_display_name()
+                ),
+            )
+
+        obj.refresh_from_db()
+        self.assertEqual(obj.status, 'ACCEPTED')
+        self._assert_alert_count(CREATE_ALERT, self.user, 0)
+        self._assert_alert_count(CREATE_ALERT, self.user_delegate, 0)
+        self._assert_alert_count(ACCEPT_ALERT, self.user, 0)
+        self._assert_alert_count(ACCEPT_ALERT, self.user_delegate, 0)
+        self.assert_irods_obj(coll_path, False)
+        self.assert_irods_obj(obj_path, False)
+
+
+class TestIrodsRequestAcceptBatchView(TestIrodsRequestViewsBase):
+    """Tests for IrodsRequestAcceptBatchView"""
+
+    def test_render(self):
+        """Test rendering IrodsRequestAcceptBatchView"""
+        self.assert_irods_obj(self.path)
+        self.assert_irods_obj(self.path_md5)
+
+        with self.login(self.user_contrib):
+            self.client.post(
```

Comment:
Since we already test the creation in its dedicated test class, you could simplify accept/reject/delete tests by replacing these creation post requests by simply creating the request object in the SODAR database. For this you can use `IrodsDataRequestMixin.make_irods_data_request()`.

Reply:
It is way better approach, but after changing the current implementation, most alert checks start failing. It won't be a problem if that would be create alerts after creation, but there are use-cases of checking create alerts after deleting, accepting and rejecting requests (it also shows different behaviour if there are multiple requests).

I see couple of possibilities to fix that:
- Avoid alert tests and just replace creations.
- Avoid direct creation method and let it be as it is.
- Write new tests for alert testing with old implementation and change every other test to new one.

What do you think on that?

Reply:
It should be enough if we test the create alerts in the create view tests. They should not be affected by delete/accept/reject requests, right? The create alerts shouldn't be e.g. modified afterwards by any further actions, if I'm correct?

There are indeed things which affect how create alerts are generated, but these should be tested in `TestIrodsRequestCreateView`. There we should, of course, use a `POST` request to create the request AND the necessary alerts, while in the subsequent test classes the creation can be simplified and we only look for the alerts specific to that view under testing.

This will also make the tests more atomic and consistent, as we specifically test the functionality of the view in question.

So my suggestion would be:

- Test all necessary create alert actions in `TestIrodsRequestCreateView`
- In test classes for accept/reject/delete views, simplify the request creation and only test for the alerts relevant to the view in question

It's possible I'm missing some case where this isn't feasible so if you disagree, please let me know.